src/controllers/companyController.js

```diff
@@ -0,0 +1,86 @@
+import { companies } from "../models/index.js";
+import NotFoundError from "../errors/notFoundError.js";
+import validateCreateCompany from "../usecases/validateCreateCompany.js";
+
+class CompanyController {
+  static async registerCompany(req, res, next) {
+    try {
+      if (validateCreateCompany(req.body)) {
+        const newCompany = await companies.create(req.body);
+        res.status(201).json({ company: newCompany });
+      }
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  static async listCompanyById(req, res, next) {
+    try {
+      const id = req.params.id;
+      const foundCompany = await companies.findById(id);
+      if (!foundCompany) {
+        next(new NotFoundError("Loja não encontrada!"));
+        return;
+      }
+      res.status(200).json(foundCompany);
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  static async updateCompany(req, res, next) {
+    try {
+      if (validateCreateCompany(req.body, true)) {
+        const id = req.params.id;
+        const updatedCompany = await companies.findByIdAndUpdate(id, req.body, {
+          runValidators: true,
+          new: true,
+        });
+        if (!updatedCompany) {
+          next(new NotFoundError("Loja não encontrada!"));
+          return;
+        }
+        res.status(200).json({ company: updatedCompany });
+      }
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  static async listCompanies(req, res, next) {
+    try {
+      const search = buildSearchQuery(req.query);
+
+      if (!search) {
+        res.status(200).send([]);
+        return;
+      }
```

Comment:
Aqui eu tenho dois pontos:

1. Considerando que o método `buildSearchQuery` irá retornar um objeto vazio se nenhum parâmetro de query for informado, esse trecho de código dentro desse `if` nunca será executado, pois se você fizer a validação `if (! {} )` o resultado é false, e não true. Para validar se um objeto é vazio, existem outras estratégias, como verificar se a quantidade de chaves do objeto é igual a zero ( `Object.keys(search).length === 0`)

2. Conforme mencionei no [comentário sobre tratamento de queries](https://github.com/fabifabufabo/estag-2/pull/1#discussion_r1699051062), o fato de eu não passar nenhuma query não significa que o retorno da API deve ser vazio. Pelo contrário, se eu não utilizo filtro, eu devo retornar todos os dados da base (levando em consideração os parâmetros de paginação e estratégias de paginação default, para também não retornar literalmente TODOS os documentos salvos). A não ser que seja uma regra de negócio (o que não é o caso dessa task), não se deve condicionar o retorno de uma API com base na condição do cliente informar um parâmetro de query ou não. O uso de queries em consulta é opcional para o cliente.

Reply:
> Considerando que o método buildSearchQuery irá retornar um objeto vazio se nenhum parâmetro de query for informado, esse trecho de código dentro desse if nunca será executado, pois se você fizer a validação if (! {} ) o resultado é false, e não true. Para validar se um objeto é vazio, existem outras estratégias, como verificar se a quantidade de chaves do objeto é igual a zero ( Object.keys(search).length === 0)

Fiz os ajustes :)

Reply:
> Conforme mencionei no https://github.com/fabifabufabo/estag-2/pull/1#discussion_r1699051062, o fato de eu não passar nenhuma query não significa que o retorno da API deve ser vazio. Pelo contrário, se eu não utilizo filtro, eu devo retornar todos os dados da base (levando em consideração os parâmetros de paginação e estratégias de paginação default, para também não retornar literalmente TODOS os documentos salvos). A não ser que seja uma regra de negócio (o que não é o caso dessa task), não se deve condicionar o retorno de uma API com base na condição do cliente informar um parâmetro de query ou não. O uso de queries em consulta é opcional para o cliente.

Feito!